### PR TITLE
Add Support For More Request/Response Headers

### PR DESCRIPTION
**Pull request by asherbiny**
Adding support for language, timezone, role, transaction-id headers.
Adding support for set-role, started-transaction-id and clear-transaction-id response headers.
Removed warning.

### CODE REVIEW

File: src/headers.rs [hunk 1]
```diff
@@ -141,14 +141,46 @@ impl Headers {
         self
     }
 
-    // TODO: time_zone
-    // TODO: language
+    /// Sets the timezone to be used when running the query, which by default is the timezone of the Presto engine.
+    /// Example: America/Los_Angeles
+    /// The `timezone` field must only contain visible ASCII characters (32-127);
```
Comment (jagill) [inline src/headers.rs:146]: Separate from this PR, I think I should extract these repetitive warnings into a top-level warning.

File: src/headers.rs [hunk 2]
```diff
@@ -364,6 +463,24 @@ mod tests {
...
+            get_value(&header_map, "x-trino-transaction-id"),
+            Some("abc123".to_string())
+        );
+        
+        response_header_map.clear();
+        response_header_map.insert("X-Trino-Clear-Transaction-Id", "".parse().unwrap());
+
+        let mut request_headers = Headers::trino().transaction_id("abc123");
+        request_headers
+        .update_from_response_headers(&response_header_map)
```
Comment (jagill) [inline src/headers.rs:480]: Did you run `cargo fmt` on this?  Indenting looks off.

**Review by jagill — APPROVED** [review]
This looks great.  One tiny comment on formatting -- if you could run `cargo fmt` and if it changes, resubmit by doing a `git push -f`.

Once that's done (or you tell me that it didn't result in a change), I'll merge this.  I'm going to merge it manually, because I like the cleaner commit history from rebasing.

**Comment by jagill** [comment]
Please also reference the issue that this PR is addressing, with `fixes #XXX` if it addresses it completely.  If partially, just note that without the `fixes` keyword.

File: src/headers.rs (hunk 1 above)
Comment (asherbiny) [reply src/headers.rs:146]: Well it's a tradeoff... I agree that the repetitive warning is a pain for maintainability. On the other hand, having the warning on top of each method will help make it visible to end-users through IDE tooltips and such. I would lean slightly towards making this fool-proof for the end users.

File: src/headers.rs (hunk 2 above)
Comment (asherbiny) [reply src/headers.rs]: Oops, I've been using company tooling for a while and forgot about `cargo fmt`. Just ran it now and fixed the formatting.

**Comment by asherbiny** [comment]
> This looks great. One tiny comment on formatting -- if you could run `cargo fmt` and if it changes, resubmit by doing a `git push -f`.
> 
> Once that's done (or you tell me that it didn't result in a change), I'll merge this. I'm going to merge it manually, because I like the cleaner commit history from rebasing.

Thanks for the review! I've updated the formatting. Please merge it when you get a chance.

**Comment by jagill** [comment]
Thanks!

**Pull request merged**